### PR TITLE
Update the docs and formatting

### PR DESCRIPTION
**Pull request by wsargent**
Play Slick has some outdated links and documentation.

* Updated to use Lightbend instead of Typesafe
* Set example to use `profile` instead of `driver`
* Update links that still pointed to 3.1.0 and fixed incorrect escaping
* Point to HikariCP github project because docs site is offline
* Point to the discourse sites instead of the google groups.

### CODE REVIEW

File: docs/manual/working/scalaGuide/main/sql/slick/PlaySlickMigrationGuide.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 This is a guide for migrating from Play Slick v0.8 to v1.0, v1.1 or v2.0.0.
```
Comment (will-sargent-eero) [inline docs/manual/working/scalaGuide/main/sql/slick/PlaySlickMigrationGuide.md]: If this is a guide for migrating to v2.0.0, why do we mention "that you have read the [Slick 3.1 documentation](http://slick.typesafe.com/docs/)" later?

File: docs/manual/working/scalaGuide/main/sql/slick/PlaySlick.md [hunk 2]
```diff
@@ -58,11 +58,11 @@ slick.dbs.default.db.driver="org.h2.Driver"
 slick.dbs.default.db.url="jdbc:h2:mem:play"
 ```
 
-First, note that the above is a valid Slick configuration (for the complete list of configuration parameters that you can use to configure a database see the Slick ScalaDoc for [Database.forConfig](http://slick.typesafe.com/doc/3.1.0/api/index.html#slick.jdbc.JdbcBackend$DatabaseFactoryDef@forConfig%28path:String,config:com.typesafe.config.Config,driver:java.sql.Driver,classLoader:ClassLoader%29:JdbcBackend.this.Database) - make sure to expand the `forConfig` row in the doc).
+First, note that the above is a valid Slick configuration (for the complete list of configuration parameters that you can use to configure a database see the Slick ScalaDoc for [Database.forConfig](http://slick.typesafe.com/doc/3.3.2/api/index.html#slick.jdbc.JdbcBackend$DatabaseFactoryDef@forConfig%28path:String,config:com.typesafe.config.Config,driver:java.sql.Driver,classLoader:ClassLoader%29:JdbcBackend.this.Database) - make sure to expand the `forConfig` row in the doc).
```
Comment (marcospereira) [inline docs/manual/working/scalaGuide/main/sql/slick/PlaySlick.md]:
```suggestion
First, note that the above is a valid Slick configuration (for the complete list of configuration parameters that you can use to configure a database see the Slick ScalaDoc for [Database.forConfig](https://scala-slick.org/doc/3.3.2/api/index.html#slick.jdbc.JdbcBackend$DatabaseFactoryDef@forConfig%28path:String,config:com.typesafe.config.Config,driver:java.sql.Driver,classLoader:ClassLoader%29:JdbcBackend.this.Database) - make sure to expand the `forConfig` row in the doc).
```

File: docs/manual/working/scalaGuide/main/sql/slick/PlaySlickAdvancedTopics.md [hunk 3]
```diff
@@ -4,19 +4,19 @@
...
 
-Also, note that as stated in the [Slick documentation](http://slick.typesafe.com/doc/3.1.0/database.html#connection-pools), a reasonable default for the connection pool size is calculated from the thread pool size. In fact, you should only need to tune `numThreads` and `queueSize` in most cases, for each of your database configuration.
+Also, note that as stated in the [Slick documentation](http://scala-slick.org/doc/3.3.2/database.html#connection-pools), a reasonable default for the connection pool size is calculated from the thread pool size. In fact, you should only need to tune `numThreads` and `queueSize` in most cases, for each of your database configuration.
```
Comment (marcospereira) [inline docs/manual/working/scalaGuide/main/sql/slick/PlaySlickAdvancedTopics.md]:
```suggestion
Also, note that as stated in the [Slick documentation](https://scala-slick.org/doc/3.3.2/database.html#connection-pools), a reasonable default for the connection pool size is calculated from the thread pool size. In fact, you should only need to tune `numThreads` and `queueSize` in most cases, for each of your database configurations.
```

File: docs/manual/working/scalaGuide/main/sql/slick/PlaySlickAdvancedTopics.md [hunk 4]
```diff
@@ -4,19 +4,19 @@
...
 
-Finally, it's worth mentioning that while Slick allows using a different connection pool than [HikariCP](http://brettwooldridge.github.io/HikariCP/) (though, Slick currently only offers built-in support for HikariCP, and requires you to provide an implementation of [JdbcDataSourceFactory](http://slick.typesafe.com/doc/3.1.0/api/index.html#slick.jdbc.JdbcDataSourceFactory) if you want to use a different connection pool), Play Slick currently doesn't allow using a different connection pool than HikariCP.
+Finally, it's worth mentioning that while Slick allows using a different connection pool than [HikariCP](https://github.com/brettwooldridge/HikariCP) (though, Slick currently only offers built-in support for HikariCP, and requires you to provide an implementation of [JdbcDataSourceFactory](http://scala-slick.org/doc/3.3.2/api/index.html#slick.jdbc.JdbcDataSourceFactory) if you want to use a different connection pool), Play Slick currently doesn't allow using a different connection pool than HikariCP.
```
Comment (marcospereira) [inline docs/manual/working/scalaGuide/main/sql/slick/PlaySlickAdvancedTopics.md]:
```suggestion
Finally, it's worth mentioning that while Slick allows using a different connection pool than [HikariCP](https://github.com/brettwooldridge/HikariCP) (though, Slick currently only offers built-in support for HikariCP, and requires you to provide an implementation of [JdbcDataSourceFactory](https://scala-slick.org/doc/3.3.2/api/index.html#slick.jdbc.JdbcDataSourceFactory) if you want to use a different connection pool), Play Slick currently doesn't allow using a different connection pool than HikariCP.
```

File: docs/manual/working/scalaGuide/main/sql/slick/PlaySlickMigrationGuide.md (hunk 1 above)
Comment (marcospereira) [reply docs/manual/working/scalaGuide/main/sql/slick/PlaySlickMigrationGuide.md]: Bad find and replace, maybe? 🤷‍♂

File: docs/manual/working/scalaGuide/main/sql/slick/code/DI.scala [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2011-2015 Typesafe Inc. <http://www.typesafe.com>
+ * Copyright (C) 2011-2019 Lightbend Inc. <https://www.lightbend.com>
```
Comment (marcospereira) [inline docs/manual/working/scalaGuide/main/sql/slick/code/DI.scala]: Could you please remove the year like we did for Play projects?

```scala
/*
 * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
 */
```

Comment (wsargent) [reply docs/manual/working/scalaGuide/main/sql/slick/code/DI.scala]: Done.

**Comment by marcospereira** [comment]
Thanks, Will!

**Pull request merged**